static/app/views/profiling/landing/table.tsx

```diff
@@ -0,0 +1,152 @@
+import {useMemo} from 'react';
+import {Location} from 'history';
+
+import DateTime from 'sentry/components/dateTime';
+import Duration from 'sentry/components/duration';
+import GridEditable, {
+  COL_WIDTH_UNDEFINED,
+  GridColumnOrder,
+} from 'sentry/components/gridEditable';
+import {IconCheckmark, IconClose} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {Trace} from 'sentry/types/profiling/trace';
+import {Container, NumberContainer} from 'sentry/utils/discover/styles';
+
+interface Props {
+  location: Location;
+  traces: Trace[];
+}
+
+function ProfileTable({location, traces}: Props) {
+  const data: TableDataRow[] = useMemo(
+    () =>
+      traces.map(trace => ({
+        failed: Boolean(trace.failed), // makes this a required field
+        ...trace,
+      })),
+    [traces]
+  );
+
+  return (
+    <GridEditable
+      isLoading={false}
+      data={data}
+      columnOrder={COLUMN_ORDER.map(key => COLUMNS[key])}
+      columnSortBy={[]}
+      grid={{renderBodyCell}}
+      location={location}
+    />
+  );
+}
+
+type TableColumnKey =
+  | 'id'
+  | 'failed'
+  | 'app_version'
+  | 'interaction_name'
+  | 'start_time_unix'
+  | 'trace_duration_ms'
+  | 'device_model'
+  | 'device_class';
+
+type TableColumn = GridColumnOrder<TableColumnKey>;
+type TableDataRow = Record<TableColumnKey, any>;
+
+const COLUMN_ORDER: TableColumnKey[] = [
+  'id',
+  'failed',
+  'app_version',
+  'interaction_name',
+  'start_time_unix',
+  'trace_duration_ms',
+  'device_model',
+  'device_class',
+];
+
+const COLUMNS: Record<TableColumnKey, TableColumn> = {
+  id: {
+    key: 'id',
+    name: t('Flamegraph'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  failed: {
+    key: 'failed',
+    name: t('Status'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  app_version: {
+    key: 'app_version',
+    name: t('Version'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  interaction_name: {
+    key: 'interaction_name',
+    name: t('Interaction Name'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  start_time_unix: {
+    key: 'start_time_unix',
+    name: t('Timestamp'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  trace_duration_ms: {
+    key: 'trace_duration_ms',
+    name: t('Duration'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  device_model: {
+    key: 'device_model',
+    name: t('Device Model'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+  device_class: {
+    key: 'device_class',
+    name: t('Device Class'),
+    width: COL_WIDTH_UNDEFINED,
+  },
+};
+
+function renderBodyCell(
```

Comment:
I generally like to inline these to keep them as close to the render pass as possible - multiple components/render functions per file can make it annoying to track control flow and we should try to keep the code as simple as possible here. Because the grid component seems to be really small ~40loc, I think we have space here to lift this up

Reply:
I can lift this higher in the file but I don't think inlining here makes sense because `renderBodyCell` isn't a component but rather a render function that's passed as a prop to the grid. Inlining would just result in a new reference to this function being recreated repeatedly.

Reply:
Yeah, I'm not a fan of it. Since components are just functions, can we maybe defined this as a <table component name>BodyCell.tsx and just pass it as `{renderBodyCell: CellComponent}`?